File: python/500-keyBoardRow.py
class Solution(object):
    def findWords(self, words):
        l1 = "qwertyuiop"
        l2 = "asdfghjkl"
        l3 = "zxcvbnm"
        l = [l1, l2, l3]
        def check(word):
            temp = word.lower()
            pos = 0
            for i in range(3):
                if temp[0] in l[i]:
                    pos = i
            return all(s in l[pos] for s in temp)
        return filter(check, words)

File: java/500-keyBoardRow.java
public class Solution {
    public String[] findWords(String[] words) {
        String[] rows = {"qwertyuiop", "asdfghjkl", "zxcvbnm"};
        Map<Character, Integer> map = new HashMap();
        for(int i = 0; i < 3; i ++){
            for (char c: rows[i].toCharArray()){
                map.put(c, i);
            }
        }
        List<String> res = new LinkedList<>();
        for (String word : words){
            if(word.equals("")) continue;
            int index = map.get(word.toLowerCase().charAt(0));
            for (char c: word.toLowerCase().toCharArray()){
                if(map.get(c) != index){
                    index = -1;
                    break;
                }
            }
            if (index!=-1) res.add(word);
        }
        return res.toArray(new String[0]);
    }
}
